eng/scripts/analyze-step-helpers.ps1

```diff
@@ -0,0 +1,100 @@
+Write-Host "$PSScriptRoot"
+. (Join-Path $PSScriptRoot .. common scripts common.ps1)
+
+<#
+.DESCRIPTION
+    Create .gosource APIVIew artifact for go
+.PARAMETER ServiceDirectory
+    Thee name of the ServiceDirectory
+.PARAMETER OutputDirectory
+    Base output Directory path for the generated gosource artifacts
+.PARAMETER DirectoryToPublish
+    Directory containing all artifacts to be publisehd to the pipeline
+#>
+function New-GoArtifacts {
+        Param(
+        [Parameter(Mandatory=$True)]
+        [string] $ServiceDirectory,
+        [Parameter(Mandatory=$True)]
+        [string] $OutputDirectory,
+        [Parameter(Mandatory=$True)]
+        [string] $DirectoryToPublish       
+    )
+
+    foreach ($sdk in (Get-AllPackageInfoFromRepo $ServiceDirectory))
+    {
+        Write-Host "Creating API review artifact for $($sdk.Name)"
+        $sdkDirectoryPath = Join-Path -Path $OutputDirectory $sdk.Name
```

Comment:
What's the value of sdk.name?  Go has same module name under different path and that's the reason it uses complete sdk path as package name. We cannot use just the name of last folder that contains go.mod to create unique package folders.

Reply:
sdk.name for tempate `"Name": "sdk/template/aztemplate"`